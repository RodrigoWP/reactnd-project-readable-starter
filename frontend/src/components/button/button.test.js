'use strict'

import React from 'react'
import { shallow, mount } from 'enzyme'
import { Button } from '.'

it('Should render correctly', () => {
  const component = (
    <Button label='default' />
  )

  expect(shallow(component)).toMatchSnapshot()
})

it('Should call onClick when button is clicked', () => {
  const onClick = jest.fn()

  const component = mount(
    <Button
      label='default'
      onClick={onClick}
    />
  )

  component.find('button').simulate('click')

  expect(onClick).toHaveBeenCalledTimes(1)
})

it('When prop icon is not null should render img element', () => {
  const component = mount(
    <Button icon='icon' />
  )

  const imgElement = component.find('img')

  expect(imgElement.length).toBe(1)
})
