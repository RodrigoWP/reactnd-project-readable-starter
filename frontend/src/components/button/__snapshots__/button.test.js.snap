// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<button
  className=""
  style={
    Object {
      "height": "40px",
      "width": "130px",
    }
  }
>
  <span>
    default
  </span>
</button>
`;
